File: Medium/2336. Smallest Number in Infinite Set/Solution.cpp
class SmallestInfiniteSet {
public:
    set<int> st;
    int cur;
    SmallestInfiniteSet() {
        cur = 1;
    }

    int popSmallest() {
        int val = *st.begin();
        if(!st.empty()) {
            st.erase(st.begin());
            return val;
        }
        cur++;
        return cur - 1;
    }

    void addBack(int num) {
        if(num < cur) st.insert(num);
    }
};

File: Medium/2336. Smallest Number in Infinite Set/Solution.java
import java.util.*;
class SmallestInfiniteSet {
    TreeSet<Integer> set;
    int cur;
    public SmallestInfiniteSet() {
        cur = 1;
        set = new TreeSet<>();
    }

    public int popSmallest() {
        if(!set.isEmpty()) {
            int val = set.first();
            set.remove(val);
            return val;
        }
        cur++;
        return cur - 1;
    }

    public void addBack(int num) {
        if(num < cur) set.add(num);
    }
}